src/vs/base/common/stream.ts

```diff
@@ -396,18 +402,30 @@ class WriteableStreamImpl<T> implements WriteableStream<T> {
 	}
 
 	private flowData(): void {
-		if (this.buffer.data.length > 0) {
+		// if buffer is empty, nothing to do
+		if (this.buffer.data.length === 0) {
+			return;
+		}
+
+		// if buffer data can be reduced into a single object,
+		// emit the reduced data
+		if (typeof this.reducer === 'function') {
```

Comment:
@legomushroom can you explain what you did here and why?

Reply:
@legomushroom I understand how, its about allowing to pass in `null` as reducer. That makes sense to me and I appreciate that you also added a test for it 👍